### PR TITLE
Use sassc instead of node-sass

### LINKED ISSUE #15068: node-sass not available on non-x86 platforms
Cockpit version: 234
OS: ALTLinux

Build on non x86_64 platforms fails with:
```
[00:00:58] ERROR in ./pkg/lib/patternfly/patternfly-cockpit.scss
[00:00:58] Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
[00:00:58] ModuleBuildError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
[00:00:58] Error: Node Sass does not yet support your current environment: Linux Unsupported architecture (arm64) with Node.js 14.x
[00:00:58] For more information on which environments are supported please see:
[00:00:58] https://github.com/sass/node-sass/releases/tag/v4.14.1
[00:00:58]     at module.exports (/usr/src/RPM/BUILD/cockpit-234/node_modules/node-sass/lib/binding.js:13:13)
[00:00:58]     at Object.<anonymous> (/usr/src/RPM/BUILD/cockpit-234/node_modules/node-sass/lib/index.js:14:35)
[00:00:58]     at Module._compile (internal/modules/cjs/loader.js:1063:30)
[00:00:58]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
[00:00:58]     at Module.load (internal/modules/cjs/loader.js:928:32)
[00:00:58]     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
[00:00:58]     at Module.require (internal/modules/cjs/loader.js:952:19)
[00:00:58]     at require (internal/modules/cjs/helpers.js:88:18)
[00:00:58]     at getDefaultSassImplementation (/usr/src/RPM/BUILD/cockpit-234/node_modules/sass-loader/dist/getDefaultSassImplementation.js:24:10)
[00:00:58]     at getSassImplementation (/usr/src/RPM/BUILD/cockpit-234/node_modules/sass-loader/dist/getSassImplementation.js:19:72)
[00:00:58]     at Object.loader (/usr/src/RPM/BUILD/cockpit-234/node_modules/sass-loader/dist/index.js:40:61)
[00:00:58]     at /usr/src/RPM/BUILD/cockpit-234/node_modules/webpack/lib/NormalModule.js:316:20
[00:00:58]     at /usr/src/RPM/BUILD/cockpit-234/node_modules/loader-runner/lib/LoaderRunner.js:367:11
[00:00:58]     at /usr/src/RPM/BUILD/cockpit-234/node_modules/loader-runner/lib/LoaderRunner.js:233:18
[00:00:58]     at runSyncOrAsync (/usr/src/RPM/BUILD/cockpit-234/node_modules/loader-runner/lib/LoaderRunner.js:143:3)
[00:00:58]     at iterateNormalLoaders (/usr/src/RPM/BUILD/cockpit-234/node_modules/loader-runner/lib/LoaderRunner.js:232:2)
[00:00:58]     at Array.<anonymous> (/usr/src/RPM/BUILD/cockpit-234/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
[00:00:58]     at Storage.finished (/usr/src/RPM/BUILD/cockpit-234/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
[00:00:58]     at /usr/src/RPM/BUILD/cockpit-234/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
[00:00:58]     at /usr/src/RPM/BUILD/cockpit-234/node_modules/graceful-fs/graceful-fs.js:123:16
[00:00:58]     at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:63:3)
[00:00:58]  @ ./pkg/apps/index.jsx 19:0-51
[00:00:58]  @ multi ./pkg/apps/index.jsx
```

As of ce60c855b7415201a20add47ba61dab7e1ecedd1 `node-sass` is the strict requirement.
According to https://github.com/sass/node-sass/releases/tag/v4.14.1

OS | Architecture | Node
-- | -- | --
Linux* | x86 & x64 | 0.10, 0.12, 1, 2, 3, 4, 5, 6, 7, 8**, 9**, 10**^, 11**^, 12**^, 13**^, 14**^

*Linux support refers to Ubuntu, Debian, and CentOS 5+
** Not available on CentOS 5
^ Only available on x64

Build from scratch is:
- install node modules
- **autoreconf**
- configure
- make

So, only x86_64 is successful.

### PR DESCRIPTION
**Pull request by KKoukiou**
node-sass is a compiled ELF module, which is problematic for
distributions that want to rebuild everything from source. The sassc CLI
program is packaged everywhere, and both use the same libsass library.

So drop node-sass and sass-loader, and replace it with a simple loader
wrapper around sassc.

This also saves 122 npm packages (16 MB in node_modules/).

Ported from starter-kit commits 233b5cfe04142, 8821db1daf1, and 50ef8a93623.

Fixes #15068

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -16,11 +16,11 @@ various development dependencies.
 
 On Debian and recent Ubuntu ≥ 19.04:
 
-    $ sudo apt-get install nodejs npm
+    $ sudo apt-get install nodejs sassc
```
Comment (martinpitt) [inline HACKING.md:19]: Please keep npm. `nodejs` is a bit redundant as `npm` depends on it. But the `apt-get` and `dnf install` commands should have the same package names for consistency.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Niiice! Just a triviality, otherwise I'm eager to see the test results.

File: HACKING.md (hunk 1 above)
Comment (martinpitt) [reply HACKING.md]: Done while I was at it.

**Comment by martinpitt** [comment]
I taught sassc-loader to respect `$SRCDIR` so that out-of-tree builds continue to work. The previous version did that with `includePaths: path.resolve(nodedir)`, this now does the moral equivalent of that.

**Comment by martinpitt** [comment]
Green! Running all tests.

**Review by martinpitt — APPROVED** [review]
Thanks for this! I'm really happy about all these cleanups.

**Comment by martinpitt** [comment]
@lnussel : This might make your downstream packaging a little easier?

**Comment by lnussel** [comment]
Sounds good at least :+1:
Will avoid having to recompile sass in the spec file of cockpit

**Pull request merged**